after update to 2.5.0~S1569 my JSONDB throws errors

After update to snapshot 1569 I get errors in the log and quite a lot of my things are missing in the setup now.

```
2019-04-13 21:23:14.070 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@d0ddad'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.074 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@1e1dc61'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.077 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@cc6576'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.081 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@8e7bac'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.085 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@f9fae5'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.channels.segments
2019-04-13 21:23:14.088 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@14c2c4f'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.092 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@a80805'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.095 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@1c25627'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.099 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@23bba8'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.102 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@dbc9f7'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.106 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@c71349'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.109 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@12b0218'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.112 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@8a18ae'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.channels.segments
2019-04-13 21:23:14.115 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@54e939'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.channels.segments
2019-04-13 21:23:14.118 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@51f261'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.channels.segments
2019-04-13 21:23:14.121 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@1cd1df1'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.126 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@1dd35a7'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.130 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@829078'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.133 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@4073ae'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.203 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@1b5bc6'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.channels[0]
2019-04-13 21:23:14.207 [ERROR] [me.storage.json.internal.JsonStorage] - Couldn't deserialize value 'org.eclipse.smarthome.storage.json.internal.StorageEntry@1722ef'. Root cause is: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.channels.segments
```

--- a/bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/BridgeImpl.java
+++ b/bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/BridgeImpl.java
@@ -15,8 +15,7 @@ package org.eclipse.smarthome.core.thing.internal;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
@@ -32,14 +31,18 @@ import org.slf4j.LoggerFactory;
 /**
  *
  * @author Denis Nobel - Initial contribution
- * @author Christoph Weitkamp - Changed internal handling of things and added method `getThing(ThingUID)`
+ * @author Christoph Weitkamp - Added method `getThing(ThingUID)`
  */
 @NonNullByDefault
 public class BridgeImpl extends ThingImpl implements Bridge {
 
     private final transient Logger logger = LoggerFactory.getLogger(BridgeImpl.class);
 
-    private final transient Map<ThingUID, Thing> things = new ConcurrentHashMap<>();
+    /*
+     * !!! DO NOT CHANGE - We are not allowed to change the members of the BridgeImpl implementation as the storage for
+     * things uses this implementation itself to store and restore the data.
+     */
+    private transient List<Thing> things = new CopyOnWriteArrayList<>();
 
     /**
      * Package protected default constructor to allow reflective instantiation.
@@ -73,21 +76,26 @@ public class BridgeImpl extends ThingImpl implements Bridge {
     }
 
     public void addThing(Thing thing) {
-        things.put(thing.getUID(), thing);
+        things.add(thing);
     }
 
     public void removeThing(Thing thing) {
-        things.remove(thing.getUID(), thing);
+        things.remove(thing);
     }
 
     @Override
     public @Nullable Thing getThing(ThingUID thingUID) {
-        return things.get(thingUID);
+        for (Thing thing : things) {
+            if (thing.getUID().equals(thingUID)) {
+                return thing;
+            }
+        }
+        return null;
     }
 
     @Override
     public List<Thing> getThings() {
-        return Collections.unmodifiableList(new ArrayList<>(things.values()));
+        return Collections.unmodifiableList(new ArrayList<>(things));
     }
 
     @Override

--- a/bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingImpl.java
+++ b/bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingImpl.java
@@ -45,16 +45,20 @@ import org.eclipse.smarthome.core.thing.binding.builder.ThingStatusInfoBuilder;
  *         ThingType Description
  * @author Thomas Höfer - Added thing and thing type properties
  * @author Simon Kaufmann - Added label
- * @author Christoph Weitkamp - Changed internal handling of things and added method `getChannel(ChannelUID)`
+ * @author Christoph Weitkamp - Added method `getChannel(ChannelUID)`
  */
 @NonNullByDefault
 public class ThingImpl implements Thing {
 
+    /*
+     * !!! DO NOT CHANGE - We are not allowed to change the members of the ThingImpl implementation as the storage for
+     * things uses this implementation itself to store and restore the data.
+     */
     private @Nullable String label;
 
     private @Nullable ThingUID bridgeUID;
 
-    private final Map<ChannelUID, Channel> channels = new HashMap<>();
+    private List<Channel> channels = new ArrayList<>();
 
     private Configuration configuration = new Configuration();
 
@@ -133,13 +137,13 @@ public class ThingImpl implements Thing {
 
     @Override
     public List<Channel> getChannels() {
-        return Collections.unmodifiableList(new ArrayList<>(this.channels.values()));
+        return Collections.unmodifiableList(new ArrayList<>(this.channels));
     }
 
     @Override
     public List<Channel> getChannelsOfGroup(String channelGroupId) {
         List<Channel> channels = new ArrayList<>();
-        for (Channel channel : this.channels.values()) {
+        for (Channel channel : this.channels) {
             if (channelGroupId.equals(channel.getUID().getGroupId())) {
                 channels.add(channel);
             }
@@ -149,13 +153,17 @@ public class ThingImpl implements Thing {
 
     @Override
     public @Nullable Channel getChannel(String channelId) {
-        ChannelUID channelUID = new ChannelUID(uid, channelId);
-        return getChannel(channelUID);
+        for (Channel channel : this.channels) {
+            if (channel.getUID().getId().equals(channelId)) {
+                return channel;
+            }
+        }
+        return null;
     }
 
     @Override
     public @Nullable Channel getChannel(ChannelUID channelUID) {
-        return this.channels.get(channelUID);
+        return getChannel(channelUID.getId());
     }
 
     @Override
@@ -189,7 +197,7 @@ public class ThingImpl implements Thing {
     }
 
     public void addChannel(Channel channel) {
-        this.channels.put(channel.getUID(), channel);
+        this.channels.add(channel);
     }
 
     public void setChannels(List<Channel> channels) {

